shipment_advice_invoice_auto_complete/models/shipment_advice.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from collections import defaultdict
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class ShipmentAdvice(models.Model):
+    _inherit = "shipment.advice"
+
+    supplier_ids = fields.Many2many(
```

Comment:
Maybe you can make it generic and move it to shipment advice module
```suggestion
    partner_ids = fields.Many2many(
```